### PR TITLE
Improve output.

### PR DESCRIPTION
**Pull request by alexeagle**
Instead of printing a line for each file, collect them as we stream and write a single command at the end.
Also use relative paths so the command is shorter.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -12,31 +15,34 @@ module.exports = {
    * http://clang.llvm.org/docs/ClangFormatStyleOptions.html#configurable-format-style-options
    */
   checkFormat: function(options) {
-    options = (typeof options === "undefined") ?
-                  {
-                    BasedOnStyle: "Google"
-                  } :
-                  options;
+    options =
```
Comment (mprobst) [inline index.js]: Call it opt_options if it's possible to be undefined (Google style)

File: index.js [hunk 2]
```diff
@@ -12,31 +15,34 @@ module.exports = {
...
+        (typeof options === "undefined") ? {BasedOnStyle: "Google"} : options;
```
Comment (mprobst) [inline index.js]: What about just "options || {BasedOnStyle: "Google"}"?

File: index.js [hunk 3]
```diff
@@ -12,31 +15,34 @@ module.exports = {
...
     var optsStr =
         (typeof options === "string") ? options : JSON.stringify(options);
```
Comment (mprobst) [inline index.js]: Why can options be a string or a JSON object or undefined? That's a bit confusing. If that really must be, could you move that to if/else statements to make the different possible combinations clearer?

File: index.js [hunk 4]
```diff
@@ -12,31 +15,34 @@ module.exports = {
...
-    return through.obj(function(file, enc, done) {
-      var actual = file.isStream() ?
-                       file.content :
-                       fs.createReadStream(file.path, {encoding: enc});
+    return through.obj(
+        function(file, enc, done) {
+          var actual = file.isStream() ?
```
Comment (mprobst) [inline index.js]: Call "actual" actualStream to make clear what it contains?

File: index.js [hunk 5]
```diff
@@ -12,31 +15,34 @@ module.exports = {
...
-                                  " is not properly formatted.");
-                      console.log("Please run clang-format -i -style='" +
-                                  optsStr + "' " + file.path);
-                    }
-                    done();
-                  });
-    });
+          streamEqual(
+              actual,
+              clangFormat(file, enc, optsStr, done), function(err, equal) {
```
Comment (mprobst) [inline index.js]: Indentation snafu, please fix.

File: index.js (hunk 1 above)
Comment (alexeagle) [inline index.js]: done

File: index.js (hunk 3 above)
Comment (alexeagle) [inline index.js]: The reasoning is in the JSDoc.
Refactored a bit without the ternary's, PTAL

File: index.js (hunk 4 above)
Comment (alexeagle) [inline index.js]: done

File: index.js (hunk 5 above)
Comment (alexeagle) [inline index.js]: done

**Comment by mprobst** [comment]
LGTM.

**Pull request merged**